### PR TITLE
feat(releasescard): add releases widget and api

### LINKED ISSUE #128: Add ReleasesCard
Suggesting feature

Add a ReleasesCard widget that can be put in the Backstage catalog overview

### PR DESCRIPTION
**Pull request by manuelstein**
## 🚨 Proposed changes

closes #128 

- add widget ReleasesCard
- add API method getReleasesSummary according to other api calls (summary type, extract data, etc)
- add explicit dependency to @types/semver (already in yarn.lock) to let the widget filter releases

## ⚙️ Types of changes

What types of changes does your code introduce? _Put an `x` in the boxes that apply_

-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Bugfix (non-breaking change which fixes an issue)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Documentation Update (if none of the other choices apply)
-   [ ] Refactor

### CODE REVIEW

**Review by antoniomuso — CHANGES_REQUESTED** [review]
Hi manuelstein, thank you for your PR :smile:. it is an awesome feature! :rocket:  I have only some tips to fix.

We use `dev` directory to test the cards. Therefore you have to add the mocked data from this URL `https://gitlab.com/api/v4/projects/10174980/releases?` in the `dev/mock-gitlab/api-v4-v15.7.0.ts`.
To show the card you also have to add the card in the `packages/gitlab/src/components/GitlabCI` component. You can test that all works right using the `yarn start` command.

File: packages/gitlab/src/components/widgets/ReleasesCard/ReleasesCard.tsx [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+ */
+export const ReleasesCard = (props: ReleasesCardProps) => {
+    const { show = 'all', limit = 6 } = props;
+    const classes = useStyles();
+    const project_id = gitlabProjectId();
+    const project_slug = gitlabProjectSlug();
+    const gitlab_instance = gitlabInstance();
+
+    const GitlabCIAPI = useApi(GitlabCIApiRef).build(
+        gitlab_instance || 'gitlab.com'
```
[the remark this answers is not in the document]
Comment (manuelstein) [reply packages/gitlab/src/components/widgets/ReleasesCard/ReleasesCard.tsx:106]: was assuming #127, should I still change it to '0'?

File: packages/gitlab/src/components/widgets/ReleasesCard/ReleasesCard.tsx [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+    if (releases.length == 0) {
+        return <></>;
+    }
+
+    return (
+        <InfoCard
+            title="Releases"
+            deepLink={{
+                link: `${project_web_url}/-/releases`,
+                title: 'go to Releases',
```
[the remark this answers is not in the document]
Comment (manuelstein) [reply packages/gitlab/src/components/widgets/ReleasesCard/ReleasesCard.tsx:193]: PeopleCard uses "go to Contributors" and "go to Owners file", but I could align the PeopleCard links, too.

File: packages/gitlab/package.json [hunk 3]
```diff
@@ -42,6 +42,7 @@
         "@material-ui/core": "^4.12.2",
         "@material-ui/icons": "^4.9.1",
         "@material-ui/lab": "4.0.0-alpha.61",
+        "@types/semver": "^7.3.12",
```
[the remark this answers is not in the document]
Comment (manuelstein) [reply packages/gitlab/package.json:45]: thanks! I'm very new at this

File: packages/gitlab/src/components/widgets/ReleasesCard/ReleasesCard.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (antoniomuso) [reply packages/gitlab/src/components/widgets/ReleasesCard/ReleasesCard.tsx:106]: Yes, you should please. This change should be done in the other PR https://github.com/immobiliare/backstage-plugin-gitlab/pull/127.

File: packages/gitlab/src/components/widgets/ReleasesCard/ReleasesCard.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (antoniomuso) [reply packages/gitlab/src/components/widgets/ReleasesCard/ReleasesCard.tsx:193]: You have right, for the moment, you can discard this suggestion. We di

File: packages/gitlab/src/components/widgets/ReleasesCard/ReleasesCard.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (manuelstein) [reply packages/gitlab/src/components/widgets/ReleasesCard/ReleasesCard.tsx:106]: the file doesn't exist there, so I can't have it changed in #127

[the remark this answers is not in the document]
Comment (manuelstein) [reply packages/gitlab/src/components/widgets/ReleasesCard/ReleasesCard.tsx]: changed as requested

File: packages/gitlab/dev/mock-gitlab/api-v4-v15.7.0.ts [hunk 4]
```diff
@@ -5089,6 +5089,983 @@ export const mockedGitlabReqToRes: Record<string, any> = {
             approvals_before_merge: null,
         },
     ],
+    'projects/10174980/releases?': [
```
Comment (antoniomuso) [inline packages/gitlab/dev/mock-gitlab/api-v4-v15.7.0.ts:5092]: Can you format the file with Prettier ? it makes the file more readable.

**Review by antoniomuso — CHANGES_REQUESTED** [review]
LGTM! Good job! Only some small things. Can you squash the commits in one and add the breaking change in the commit as follow? : 
```
feat(releases-card): add releases widget

BREAKING CHANGE: GitlabCIApi type has the new method getReleaseSummary
```

File: packages/gitlab/dev/mock-gitlab/api-v4-v15.7.0.ts (hunk 4 above)
Comment (manuelstein) [reply packages/gitlab/dev/mock-gitlab/api-v4-v15.7.0.ts:5092]: done

**Review by antoniomuso — APPROVED** [review]

**Review by antoniomuso — APPROVED** [review]
LGTM, thank you :rocket:.

**Comment by manuelstein** [comment]
Thx, I'll update #127

**Pull request merged**